examples/file_inputformat.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <vector>
+
+#include "base/log.hpp"
+#include "base/serialization.hpp"
+#include "core/engine.hpp"
+#include "io/input/file_inputformat.hpp"
+
+/**
+ * Usage:
+ * Create objects of NFSFileInputFormat or HDFSFileInputFormat to use:
+ * ```c++ 
+ * NFS: NFSFileInputFormat infmt; infmt.set_input("/path/to/data/on/nfs");
+ * HDFS: HDFSFileInputFormat infmt; infmt.set_input("/path/to/data/on/hdfs");
+ * 
+ * husky::load(infmt, [](husky::base::BinStream& file) { ... });
+ * ```
+ * Or use unified API FileInputFormat:
+ * ```c++
+ * NFS: FileInputFormat infmt("nfs:///path/to/data/on/nfs");
+ * HDFS: FileInputFormat infmt("hdfs:///path/to/data/on/hdfs");
+ * ```
+ * Regex filter can be used to ignore files that don't match the pattern.
+ * Regex expression follows the path and should starts with a colon:
+ * ```c++
+ * FileInputFormat infmt("hdfs:///path/to/data:part-\\d+");
+ * ```
+ */
+
+void read_bin_file() {
+    husky::io::FileInputFormat infmt(husky::Context::get_param("basedir"));
+
+    size_t read_sz = 0;
+    husky::load(infmt, [&read_sz](husky::base::BinStream& bin) {
+        read_sz += bin.size();
+    });
+
+    husky::base::log_msg(std::to_string(husky::Context::get_global_tid())
+        + " reads " + std::to_string(read_sz) + " bytes.");
+}
+
+int main(int argc, char** argv) {
```

Comment:
I am doubt if I can read external files when performing the gtest? If to do a test, I have to create some files before the test part (this can be done in the constructor of the gtest class) and destroy the files after the test (done in the destructor). Right?